packages/xstate-react/src/useInterpret.ts

```diff
@@ -50,7 +51,20 @@ export function useInterpret<
     | ((value: State<TContext, TEvent, any, TTypestate>) => void)
 ): Interpreter<TContext, any, TEvent, TTypestate> {
   const machine = useConstant(() => {
-    return typeof getMachine === 'function' ? getMachine() : getMachine;
+    const machineToReturn =
+      typeof getMachine === 'function' ? getMachine() : getMachine;
+
+    if (!machineToReturn.config.on) {
+      (machineToReturn as any).config.on = {};
+    }
+
+    (machineToReturn.config as any).on['xstate.update.syncToContext'] = {
+      actions: assign((_context, event: any) => {
+        return event.props;
+      })
+    };
```

Comment:
Not sure how I feel about an option changing the actual machine config :thinking: this seems like something easy to add but is going to be difficult to remove later, and also might be something unexpected for the user inspecting their machine.

I don't have any immediate better ideas right now, though, but let's keep thinking about this and not merge it in just yet.

Reply:
I agree, it feels hacky but it appears to work nicely.

@Andarist What do you think?